cp_graph.py

```diff
@@ -790,6 +811,69 @@ def _add_output_connections(
 
 
 # ----- GRAPH FORMATTING AND OUTPUT -----
+def apply_liveness_styling(
+    G: nx.DiGraph, modules_info: List[ModuleInfo], dependency_data: Dict[str, Any]
+) -> None:
+    """
+    Apply liveness-based styling to edges in the graph.
+
+    Args:
+        G: The NetworkX graph
+        modules_info: List of module information
+        dependency_data: Original dependency graph data with liveness info
+    """
+    # Build a mapping from module_num to liveness data
+    module_liveness = {}
+    for module_data in dependency_data.get("modules", []):
+        module_num = module_data.get("module_num")
+        module_liveness[module_num] = {
+            "live": module_data["live"],
+            "disposed": module_data["disposed"],
+        }
+
+    # Apply styling to edges
+    for edge in G.edges(data=True):
+        src, dst, _ = edge
+        src_type = G.nodes[src].get("type")
+        dst_type = G.nodes[dst].get("type")
+
+        if src_type == NODE_TYPE_MODULE and dst_type in (
+            NODE_TYPE_IMAGE,
+            NODE_TYPE_OBJECT,
+        ):
+            module_node = src
+            data_node = dst
+        elif (
+            src_type in (NODE_TYPE_IMAGE, NODE_TYPE_OBJECT)
+            and dst_type == NODE_TYPE_MODULE
+        ):
+            module_node = dst
+            data_node = src
+        else:
+            continue
+
+        data_name = G.nodes[data_node].get("name")
+        if not data_name:
+            continue
+
+        module_num = G.nodes[module_node].get("module_num")
+
+        if not module_num or module_num not in module_liveness:
+            continue
+
+        liveness_data = module_liveness[module_num]
+
+        # Apply styling based on liveness
+        if data_name in liveness_data["disposed"]:
```

Comment:
What happens if data appears in both lists? The last one (live/green) wins. Consider documenting this or making it explicit.